### PR TITLE
Feat/prismic graphql exhibitions and events

### PR DESCRIPTION
**Pull request by melanierogan**
## Who is this for?

Anyone who wants to search for exhibitions or events. This covers the rest of https://github.com/wellcomecollection/wellcomecollection.org/issues/8712 and https://github.com/wellcomecollection/wellcomecollection.org/issues/8801

## What is it doing for them?

Now implements the queries needed to search events and exhibitions, returns the exhibitions in a `ResultList`. This also organises the folder structure to echo the Prismic folders in content. Again, this is currently just output as json in `<code>` tags, as this very much represents the backend side of this work.

### CODE REVIEW

File: catalogue/webapp/services/prismic/fetch/events.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+};
+
+export async function getEvents({
+  query,
+  pageSize,
+}: PrismicQueryProps): Promise<PrismicResultsList<Event> | PrismicApiError> {
+  try {
+    const res = await prismicGraphQLClient('events', query, pageSize);
+    const { allEventss } = await res;
+    const exhibitions = await transformPrismicResponse(allEventss);
```
Comment (rcantin-w) [inline catalogue/webapp/services/prismic/fetch/events.ts:19]: Should this be `events`?

But I'm also wondering if it couldn't all be just one function, we only change the first passed parameter in `prismicGraphQLClient`, right?

File: catalogue/webapp/pages/search/exhibitions.tsx [hunk 2]
```diff
@@ -5,14 +5,45 @@ import { getServerData } from '@weco/common/server-data';
 import Space from '@weco/common/views/components/styled/Space';
 import { NextPageWithLayout } from '@weco/common/views/pages/_app';
 import { getSearchLayout } from 'components/SearchPageLayout/SearchPageLayout';
+import { getExhibitions, PrismicQueryProps } from '../../services/prismic/fetch/exhibitions';
 
-export const SearchPage: NextPageWithLayout = () => {
+type Props = {
+  exhibitionResponseList;
```
Comment (rcantin-w) [inline catalogue/webapp/pages/search/exhibitions.tsx:11]: Could this be typed with the `Exhibition` from `../../types/exhibition`?

File: catalogue/webapp/pages/search/events.tsx [hunk 3]
```diff
@@ -5,14 +5,48 @@ import { getServerData } from '@weco/common/server-data';
...
 import { NextPageWithLayout } from '@weco/common/views/pages/_app';
 import { getSearchLayout } from 'components/SearchPageLayout/SearchPageLayout';
+import {
+  getEvents,
+  PrismicQueryProps,
+} from '../../services/prismic/fetch/events';
 
-export const SearchPage: NextPageWithLayout = () => {
+type Props = {
+  eventResponseList;
```
Comment (rcantin-w) [inline catalogue/webapp/pages/search/events.tsx:14]: This too could be typed with the `Event` type created in this PR?

File: catalogue/webapp/services/prismic/fetch/index.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
+import * as prismic from '@prismicio/client';
+import fetch from 'node-fetch';
+import { gql, GraphQLClient } from 'graphql-request';
+import { PrismicApiError } from '../types';
+import { capitalize } from '@weco/common/utils/grammar';
+
+export type GetServerSidePropsPrismicClient = {
```
Comment (rcantin-w) [inline catalogue/webapp/services/prismic/fetch/index.ts:7]: I don't think this is used anywhere within this project, but should it be?

File: catalogue/webapp/services/prismic/fetch/index.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  client: prismic.Client;
+};
+
+export const typesToPrismicGraphQLSchemaTypes = {
+  // types to graphql query schema types,
+  events: 'allEventss',
+  exhibitions: 'allExhibitionss',
+  articles: 'allArticless',
+  series: 'allSeriess',
+  webcomics: 'allWebcomicss',
```
Comment (rcantin-w) [inline catalogue/webapp/services/prismic/fetch/index.ts:18]: Oh, question, should we consider series and webcomics as part of "stories"? Still wrapping my head around that.

**Comment by rcantin-w** [comment]
This is so clean and easy to understand - very excited about this piece of work 🎉 I've left a few questions for my own learning and a few comments, but it looks great.

File: catalogue/webapp/services/prismic/fetch/index.ts (hunk 5 above)
Comment (melanierogan) [reply catalogue/webapp/services/prismic/fetch/index.ts:18]: Yep, I was starting the work to get these in when I looked at the original content/prismic fetch function for articles

File: catalogue/webapp/services/prismic/fetch/events.ts (hunk 1 above)
Comment (melanierogan) [reply catalogue/webapp/services/prismic/fetch/events.ts:19]: Yep! I've got some commits to push shortly that should see this working

File: catalogue/webapp/services/prismic/fetch/index.ts (hunk 4 above)
Comment (melanierogan) [reply catalogue/webapp/services/prismic/fetch/index.ts:7]: No, I was going to use something from content/services/prismic but ended up changing my mind

File: catalogue/webapp/services/prismic/fetch/events.ts (hunk 1 above)
Comment (rcantin-w) [reply catalogue/webapp/services/prismic/fetch/events.ts:19]: OK I'll hold! 👏 🙌

File: catalogue/webapp/services/prismic/fetch/index.ts (hunk 5 above)
Comment (melanierogan) [reply catalogue/webapp/services/prismic/fetch/index.ts:13]: Very good shout. As per our chat and the work on 'overview' ticket, I'm going to tackle this in another PR.

File: catalogue/webapp/services/prismic/fetch/index.ts [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  );
+  const graphQLQuery = prismicGraphQLQuery(type, query, pageSize);
+  return graphqlClient.request(graphQLQuery);
+}
+
+export const prismicApiError = (): PrismicApiError => ({
+  errorType: 'http',
+  httpStatus: 500,
+  label: 'Internal Server Error',
+  description: '',
```
Comment (rcantin-w) [inline catalogue/webapp/services/prismic/fetch/index.ts:83]: As a non blocker sidenote; would we want a description here? This displays on the FE right, so maybe not, do we have a 500 page? 🤔

File: catalogue/webapp/services/prismic/transformers/index.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+      id,
+      title: title[0]?.text,
+      image: {
+        url: image.image?.url,
+      },
+      url: `https://wellcomecollection.org/articles/${id}`,
+      firstPublicationDate,
+      contributors,
+      type: type,
+      summary: standfirst?.text[0]?.text,
```
Comment (rcantin-w) [inline catalogue/webapp/services/prismic/transformers/index.ts:23]: 🎉

File: catalogue/webapp/services/prismic/fetch/index.ts (hunk 6 above)
Comment (melanierogan) [reply catalogue/webapp/services/prismic/fetch/index.ts:83]: That's a really good point, I'll make a note to have a look at what we do right now, I'm fairly sure there's something we can just use.

File: catalogue/webapp/services/prismic/types/exhibition.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+export type Title = {
+  text: {
+    text: RichTextField;
+  };
+};
+
+export type Standfirst = {
+  text: {
+    text: string;
```
Comment (rcantin-w) [inline catalogue/webapp/services/prismic/types/exhibition.ts:25]: Could this be simplified? (will we ever need more than one?)

File: catalogue/webapp/services/prismic/types/event.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  type: ContentType[];
+};
+
+export type Title = {
+  text: {
+    text: RichTextField;
+  };
+};
+
+export type Standfirst = {
```
Comment (rcantin-w) [inline catalogue/webapp/services/prismic/types/event.ts:23]: Title and Standfirst are repeated throughout, maybe they could be in the index file? (But also maybe we wait until we're happy with all of them before simplified and I hold my horses)

File: catalogue/webapp/services/prismic/fetch/index.ts (hunk 6 above)
Comment (rcantin-w) [reply catalogue/webapp/services/prismic/fetch/index.ts:83]: Can defo be a side ticket, but something that could be useful to have!

**Review by rcantin-w — APPROVED** [review]

**Pull request merged**